### PR TITLE
Modify rule S107: Update code example for csharp to include rule behavior in the case of constructors.

### PR DESCRIPTION
**Pull request by mary-georgiou-sonarsource**
Update code example for rule S107 to reflect the case where the parameters of a constructor intended for the base class do not count in the total sum of parameters.

### CODE REVIEW

**Review by arseniy-sonar — CHANGES_REQUESTED** [review]
@mary-georgiou-sonarsource , please update the PR title according to our [naming convention](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule), and also add the `dotnet` label before merging.

**Review by arseniy-sonar — APPROVED** [review]
Thank you.

File: rules/S107/csharp/rule.adoc [hunk 1]
```diff
@@ -16,12 +16,38 @@ public void doSomething(int param1, int param2, int param3, string param4, long
...
+	}
+}
+
+public TestClass(int param1, int param2, int param3, string param4, long param5)
+{
+	public void TestClass(int param1, int param2, int param3, string param4, long param5) // Noncompliant
+	{
+	...
+	}
+}
```
Comment (andrei-epure-sonarsource) [inline rules/S107/csharp/rule.adoc:48]: I don't understand what `TestClass` is ? what's the return value of the function?

and what's the inner `TestClass`?

**Review by andrei-epure-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S107/csharp/rule.adoc (hunk 1 above)
Comment (mary-georgiou-sonarsource) [reply rules/S107/csharp/rule.adoc:48]: Apologies, the keyword `void` is a "typo".
My understanding is to leave only the first part of the example with the comment?
I agree the non-compliant example is too much.

File: rules/S107/csharp/rule.adoc [hunk 2]
```diff
@@ -16,12 +16,30 @@ public void doSomething(int param1, int param2, int param3, string param4, long
...
 {
 ...
 }
 ----
 
+== Exceptions
+
+[source,csharp]
+----
+public BaseClass(int param1)
```
Comment (andrei-epure-sonarsource) [inline rules/S107/csharp/rule.adoc:29]:
```suggestion
public class BaseClass(int param1)
```

File: rules/S107/csharp/rule.adoc [hunk 3]
```diff
@@ -16,12 +16,30 @@ public void doSomething(int param1, int param2, int param3, string param4, long
...
+{
+...
+}
+
+public ExtensionClass : BaseClass
```
Comment (andrei-epure-sonarsource) [inline rules/S107/csharp/rule.adoc:34]:
```suggestion
public class ExtensionClass : BaseClass
```

**Review by andrei-epure-sonarsource — CHANGES_REQUESTED** [review]

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM.

**Pull request merged**